From vijaykumar@bravegnu.org  Mon Oct 26 15:01:52 2009
From: vijaykumar@bravegnu.org
Date: Mon, 21 Sep 2009 11:23:54 +0530
Subject: Staging: poch: Parameter to enable synthetic counter
To: greg@kroah.com
Cc: jayakumar.lkml@gmail.com, alexey.zaytsev@gmail.com
Message-ID: <20090921055913.747374338@bravegnu.org>

From: Vijay Kumar B. <vijaykumar@bravegnu.org>

Adds a parameter that causes the hardware to synthesize Rx values
using a counter.

Signed-off-by: Vijay Kumar B. <vijaykumar@bravegnu.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/poch/poch.c |   13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

--- a/drivers/staging/poch/poch.c
+++ b/drivers/staging/poch/poch.c
@@ -245,6 +245,11 @@ struct poch_dev {
 	struct device *dev;
 };
 
+static int synth_rx;
+module_param(synth_rx, bool, 0600);
+MODULE_PARM_DESC(synth_rx,
+		"Synthesize received values using a counter. Default: No");
+
 static dev_t poch_first_dev;
 static struct class *poch_cls;
 static DEFINE_IDR(poch_ids);
@@ -827,9 +832,11 @@ static int poch_open(struct inode *inode
 			  fpga + FPGA_TX_CTL_REG);
 	} else {
 		/* Flush RX FIFO and output data to cardbus. */
-		iowrite32(FPGA_RX_CTL_CONT_CAP
-			  | FPGA_RX_CTL_FIFO_FLUSH,
-			  fpga + FPGA_RX_CTL_REG);
+		u32 ctl_val = FPGA_RX_CTL_CONT_CAP | FPGA_RX_CTL_FIFO_FLUSH;
+		if (synth_rx)
+			ctl_val |= FPGA_RX_CTL_SYNTH_DATA;
+
+		iowrite32(ctl_val, fpga + FPGA_RX_CTL_REG);
 	}
 
 	atomic_inc(&channel->inited);
